include/swift/Runtime/Mutex.h

```diff
@@ -75,8 +75,8 @@ class ScopedNotifyAllT {
 /// A ConditionVariable that works with Mutex to allow -- as an example --
 /// multi-threaded producers and consumers to signal each other in a safe way.
 class ConditionVariable {
-  friend class Mutex;
-  friend class StaticMutex;
+  friend class ConditionMutex;
+  friend class StaticConditionMutex;
```

Comment:
I wonder if there is a better name for these.

Reply:
Update: discussed offline with @mikeash and we ended up coming up with `ConditionVariable::Mutex` for the "special" mutex here.

Reply:
I made that change and nested a few other things that made sense. How's it look now?